sw/device/lib/dif/dif_alert_handler.h

```diff
@@ -0,0 +1,660 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_DIF_DIF_ALERT_HANDLER_H_
+#define OPENTITAN_SW_DEVICE_LIB_DIF_DIF_ALERT_HANDLER_H_
+
+/**
+ * @file
+ * @brief <a href="/hw/ip/alert_handler/doc/">Alert Handler</a> Device Interface
+ * Functions
+ */
+
+#include <stddef.h>
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_warn_unused_result.h"
+
+// Header Extern Guard (so header can be used from C and C++)
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+/**
+ * Generic enable/disable enumeration.
+ */
+typedef enum dif_alert_handler_enable {
+  kDifAlertHandlerEnabled,
+  kDifAlertHandlerDisabled,
+} dif_alert_handler_enable_t;
+
+/**
+ * An alert class.
+ *
+ * An alert class roughly specifies how to deal with an alert. The class
+ * determines which interrupt handler is fired for an alert, as well as the
+ * fine-grained details of the escalation policy, for when the processor
+ * fails to respond to an alert quickly enough.
+ *
+ * Alert classes serve as the alert handler's IRQ types. There is one IRQ for
+ * each class. Whenever an alert fires, the corresponding class's IRQ is
+ * serviced by the processor (if enabled).
+ */
+typedef enum dif_alert_handler_class {
+  kDifAlertHandlerClassA,
+  kDifAlertHandlerClassB,
+  kDifAlertHandlerClassC,
+  kDifAlertHandlerClassD,
+} dif_alert_handler_class_t;
```

Comment:
Correction: they're parametric at the peripheral level, and tbh it'd be ok to have a `_Static_assert` to ensure the parameter matches at the moment :)

Reply:
Last time I spoke to the HW folks they seemed to believe the parametricity wasn't going to change. I'm totally in favor of making it totally parametric instead of having the static assert, though.

Reply:
Static assert, please. Spending time parameterising stuff, esp if the hardware people don't think it's going to change, seems premature.

Reply:
If you say so. Added.